src/services/goals.js

```diff
@@ -2389,3 +2390,314 @@ export async function destroyGoal(goalIds) {
     return 0;
   }
 }
+
+/**
+ * Given a list of goal statuses, determine the final status
+ * based on the criteria provided by OHS. Intended only
+ * to be used for merge goals
+ *
+ * Note that this should be kept in sync with the status predicted by
+ * the React component the user sees (FinalGoalCard)
+ *
+ * @param {string[]} statuses
+ * @returns string one of Object.values(GOAL_STATUS)
+ */
+export function determineMergeGoalStatus(statuses) {
+  if (statuses.includes(GOAL_STATUS.IN_PROGRESS)) {
+    return GOAL_STATUS.IN_PROGRESS;
+  }
+
+  if (statuses.includes(GOAL_STATUS.CLOSED)) {
+    return GOAL_STATUS.CLOSED;
+  }
+
+  if (statuses.includes(GOAL_STATUS.SUSPENDED)) {
+    return GOAL_STATUS.SUSPENDED;
+  }
+
+  if (statuses.includes(GOAL_STATUS.NOT_STARTED)) {
+    return GOAL_STATUS.NOT_STARTED;
+  }
+  return GOAL_STATUS.DRAFT;
+}
+
+/**
+ *
+ * @param {*} objective
+ * @param {*} parentGoalId
+ */
+export async function mergeObjectiveFromGoal(objective, parentGoalId) {
+  const { dataValues } = objective;
+  const { id, goalId, ...data } = dataValues;
+  const newObjective = await Objective.create({
+    ...data,
+    goalId: parentGoalId,
+    mapsToParentObjectiveId: id,
+  }, { individualHooks: true });
+
+  const updatesToRelatedModels = [];
+
```

Comment:
Going to have to remember to update the "merge" code anytime we add goal or objective tables.

Reply:
true